### PR TITLE
Fix text formatting and code examples according to Lua style guide

### LINKED ISSUE #2236: [1pt] feedback: Module key_def | Tarantool
> <…>, fieldno = 3},
> >                           {type = 'unsigned', **|fieldno =1 }})|**
> ---
> ...
> 
> tarantool> k:extract_key({1, 99.5, 'X', nil, 99.5})
> --<…>

https://www.tarantool.io/ru/doc/latest/reference/reference_lua/key_def/#key-def-extract-key

missed whitespace after fieldno

### PR DESCRIPTION
**Pull request by NickVolynkin**
Resolve #2236

### CODE REVIEW

File: doc/reference/reference_lua/key_def.rst [hunk 1]
```diff
@@ -170,18 +171,18 @@ to extract or compare the index key values.
...
     .. _key_def-merge:
 
     .. method:: merge (other_key_def_object)
 
-        Combine the main key_def_object with other_key_def_object.
-        The return value is a new key_def_object containing all the fields of
-        the main key_def_object, then all the fields of other_key_def_object which
-        are not in the main key_def_object.
+        Combine the main ``key_def_object`` with ``other_key_def_object``.
+        The return value is a new ``key_def_object`` containing all the fields of
```
Comment (patiencedaur) [inline doc/reference/reference_lua/key_def.rst:183]: @Totktonada 
The names of the new and old objects are the same now, so it's a little confusing. The function returns a totally new object, not the altered old one—is that correct?

File: doc/reference/reference_lua/key_def.rst [hunk 2]
```diff
@@ -191,7 +192,7 @@ to extract or compare the index key values.
 
         .. code-block:: none
 
-           -- This will return a key definition with fieldno=3 and fieldno=1.
+           -- This will return a key definition with fieldno = 3 and fieldno = 1.
```
Comment (patiencedaur) [inline doc/reference/reference_lua/key_def.rst:195]: @Totktonada 
"возвращает определение ключа **по** полям fieldno = 3 и fieldno = 1"? "**для** полей? "содержащее поля"?

File: doc/reference/reference_lua/key_def.rst (hunk 1 above)
Comment (patiencedaur) [reply doc/reference/reference_lua/key_def.rst:183]: Reply from @ Totktonada:
Да, :merge() возвращает новый объект.

File: doc/reference/reference_lua/key_def.rst (hunk 2 above)
Comment (patiencedaur) [reply doc/reference/reference_lua/key_def.rst:195]: Explanation from @ Totktonada:

Давай я в целом про таплы, ключи и определения ключей напишу — должно стать понятней.

Тапл — это массив, строка в спейсе (можно сказать, в таблице, но таплы могут различаться по длине). Возьмем для примера [1, 'foo', 'bar'].

Определение ключа (key definition, key def) — это информация в духе «третье и первое поля тапла» (плюс еще типы в этих полях и еще разные детали). Порядок имеет значение.

Ключ или значение ключа — это в данном случае, собственно, массив из значения третьего и первого поля. Для тапла [1, 'foo', 'bar'] это будет ['bar', 1].

[Illustration](https://github.com/tarantool/tarantool/blob/9a1a9f09869d9dfcccb433954b4e75e3ca7d130d/test/app-tap/module_api.c#L843-L853)

File: doc/reference/reference_lua/key_def.rst [hunk 3]
```diff
@@ -19,17 +19,18 @@ to extract or compare the index key values.
 
     :param table parts: field numbers and types.
                         There must be at least one part and it
-                        must have at least fieldno and type.
+                        must have at least ``fieldno`` and ``type``.
```
Comment (patiencedaur) [inline doc/reference/reference_lua/key_def.rst:22]: @Totktonada 
Must all parts have the ``fieldno`` and ``type`` attributes? Or can there be only one part with those attributes, while the other parts don't have those attributes?

Comment (Totktonada) [reply doc/reference/reference_lua/key_def.rst:22]: All parts must have `fieldno` or `field` attribute (see #962). All parts must have `type` attribute. Other attributes are optional. The `parts` array should have at least one item.

**Comment by patiencedaur** [comment]
@Totktonada 

[Lines 11–12](https://github.com/tarantool/doc/pull/2249/files#diff-8d597cea6e6296c4eb60b7e350957f310c460c4d12fa6285991fce8d4c3b1827R11-R12):
The definition is usually used in conjunction with an index definition
to extract or compare the **index key values**.

What does "index key values" mean here? How are index and key related in this context?

**Comment by Totktonada** [comment]
Say, we have a space (named 'users'):

| id | email | name |
| --- | --- | --- |
| 1 | foo@bar.baz | Ivan Petrov |
| 2 | baz@foo.bar | Petr Ivanov |

With primary index by the 'id' field and secondary index by 'email' and 'name' fields.

A get request by the primary index could be `box.space.users:get({1})` — here `{1}` is the key that corresponds the 'id field. If we want to perform a get request using the secondary index, we should pass a key that corresponds its fields: `box.space.users:get({'foo@bar.baz', 'Ivan Petrov'})`.

So index is usually constructed on a subset of tuple's fields and phrases like 'index key' mean a key that is suitable to perfrom a request using this index.

**Pull request merged**